VMEncryption/main/EncryptionSettingsUtil.py

```diff
@@ -0,0 +1,330 @@
+#!/usr/bin/env python
+#
+# Copyright (c) Microsoft Corporation
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import httplib
+import json
+import os
+import os.path
+import socket
+import re
+import time
+import base64
+
+from shutil import copyfile
+import uuid
+from Common import CommonVariables
+
+
+class EncryptionSettingsUtil(object):
+    """ Provides capability to update encryption settings via wire server """
+
+    def __init__(self, logger):
+        self.logger = logger
+        self._DISK_ENCRYPTION_DATA_VERSION_V4 = "4.0"
+        self._DISK_ENCRYPTION_DATA_VERSION_V3 = "3.0"
```

Comment:
> self._DISK_ENCRYPTION_DATA_VERSION_V3 = "3.0" [](start = 7, length = 46)

this should not be getting used any more. Remove this and any references to it.

Reply:
Currently getting an error whilst using this extension, could this be related? 

Error: Code="VMExtensionProvisioningError" Message="VM has reported a failure when processing extension 'AzureDiskEncryptionForLinux'. Error message: \"Failed to encrypt data volumes with error: 'EncryptionSettingsUtil' object has no attribute '_DISK_ENCRYPTION_DATA_VERSION_V3', stack trace: Traceback (most recent call last):\n  File \"/var/lib/waagent/Microsoft.Azure.Security.AzureDiskEncryptionForLinux-1.1.0.48/main/handle.py\", line 1732, in daemon_encrypt\n    os_items_to_stamp=os_items_to_stamp):\n  File \"/var/lib/waagent/Microsoft.Azure.Security.AzureDiskEncryptionForLinux-1.1.0.48/main/handle.py\", line 1940, in daemon_encrypt_data_volumes\n    os_items_to_stamp=os_items_to_stamp)\n  File \"/var/lib/waagent/Microsoft.Azure.Security.AzureDiskEncryptionForLinux-1.1.0.48/main/handle.py\", line 1557, in enable_encryption_all_in_place\n    encryption_config=encryption_config)\n  File \"/var/lib/waagent/Microsoft.Azure.Security.AzureDiskEncryptionForLinux-1.1.0.48/main/handle.py\", line 193, in stamp_disks_with_settings\n    settings.post_to_wireserver(data)\n  File \"/var/lib/waagent/Microsoft.Azure.Security.AzureDiskEncryptionForLinux-1.1.0.48/main/EncryptionSettingsUtil.py\", line 274, in post_to_wireserver\n    data_copy[\"DiskEncryptionDataVersion\"] = self._DISK_ENCRYPTION_DATA_VERSION_V3\nAttributeError: 'EncryptionSettingsUtil' object has no attribute '_DISK_ENCRYPTION_DATA_VERSION_V3'\n\"\r\n\r\nMore information on troubleshooting is available at https://aka.ms/vmextensionlinuxtroubleshoot "